contracts/dtoken/src/user_profile.rs

```diff
@@ -0,0 +1,130 @@
+use near_sdk::collections::LookupMap;
+use crate::*;
+
+#[derive(BorshSerialize)]
+pub enum Action {
+    Supply,
+    Borrow,
+}
+
+#[near_bindgen]
+#[derive(BorshDeserialize, BorshSerialize)]
+pub struct UserProfile {
+    ///  Supply user amount
+    pub(crate) supply_amount: WBalance,
+
+    ///  Borrow user amount
+    pub(crate) borrow_amount: WBalance,
+}
+
+#[near_bindgen]
+#[derive(BorshDeserialize, BorshSerialize)]
+pub struct UserProfileDtoken {
```

Comment:
@TymofiiSavinov we don't need a separate wrapper for user profiles, just make `user_profile_list` a part of dtoken